pkg/pod/admission/limit_request.go

```diff
@@ -0,0 +1,98 @@
+package admission
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+
+	"k8s.io/kubernetes/pkg/admission"
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/resource"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+
+	"github.com/ghodss/yaml"
+	"speter.net/go/exp/math/dec/inf"
+)
+
+func init() {
+	admission.RegisterPlugin("PodLimitRequest", func(c kclient.Interface, config io.Reader) (admission.Interface, error) {
+		return newPodLimitRequest(config)
+	})
+}
+
+type limitRequestConfig struct {
+	Enabled                   bool    `json:"enabled"`
+	LimitCpuToMemoryRatio     float64 `json:"limitCpuToMemoryRatio"`
+	CpuRequestToLimitRatio    float64 `json:"cpuRequestToLimitRatio"`
+	MemoryRequestToLimitRatio float64 `json:"memoryRequestToLimitRatio"`
+}
+type podLimitRequest struct {
+	*admission.Handler
+	limitRequestConfig
+}
+
+// newPodLimitRequest returns an admission controller for containers that
+// configurably overrides container resource request/limits
+func newPodLimitRequest(config io.Reader) (admission.Interface, error) {
+	parsed, err := ReadConfig(config)
+	return &podLimitRequest{
+		Handler:            admission.NewHandler(admission.Create, admission.Update),
```

Comment:
what would this do on update? if it changes anything on the podspec, won't that fail to validate, since the spec is immutable?